.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+on:
+  push:
+    paths-ignore:
+      - 'CHANGELOG.md'
+    branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: '0 11 * * 4'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pre-commit:
+    uses: ./.github/workflows/step_pre-commit.yml
+
+  codeql:
+    needs: pre-commit
+    uses: ./.github/workflows/step_static-analysis.yml
+
+  test-pip:
+    needs: pre-commit
+    uses: ./.github/workflows/step_tests-pip.yml
+
+  test-conda:
+    needs: pre-commit
+    uses: ./.github/workflows/step_tests-conda.yml
+
+  build:
+    needs: [codeql, test-pip, test-conda]
+    uses: ./.github/workflows/step_build.yml
```

Comment:
Add a `re-actors/alls-green@release/v1` job to better visiualize experimental/required jobs. Here's an example:
https://github.com/swig/swig/pull/2647/files#diff-931bdf642388776529d5aeae22e23d9da62ffdadf2749c2d7aa7b4ce9c59b4a1R166-R174